corporate/lib/remote_billing_util.py

```diff
@@ -128,12 +129,13 @@ def get_remote_realm_from_session(
     return remote_realm
 
 
-def get_remote_server_from_session(
+def get_remote_server_and_user_from_session(
     request: HttpRequest,
     server_uuid: str,
-) -> RemoteZulipServer:
-    identity_dict: Optional[LegacyServerIdentityDict] = get_identity_dict_from_session(
-        request, realm_uuid=None, server_uuid=server_uuid
+) -> Tuple[RemoteZulipServer, Optional[RemoteServerBillingUser]]:
+    identity_dict = cast(
+        Optional[LegacyServerIdentityDict],
+        get_identity_dict_from_session(request, realm_uuid=None, server_uuid=server_uuid),
     )
```

Comment:
We generally aim to never use `cast` in favor of just declaring the types correctly; but we can fix that in a follow-up. @andersk FYI.